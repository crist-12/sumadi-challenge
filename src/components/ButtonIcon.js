import { View, Text, StyleSheet } from 'react-native'
import React from 'react'
import { Button } from '@ui-kitten/components'

const ButtonIcon = ({status, accessoryRight, accesoryLeft, text, appearance, callback}) => {
  return (
    <Button style={styles.buttonStyles} 
    status={status} 
    appearance={appearance} 
    accessoryRight={accessoryRight} 
    accesoryLeft={accesoryLeft} 
    onPress={callback}
    >
      {text}
    </Button>
  )
}

export default ButtonIcon;

const styles = StyleSheet.create({
  buttonStyles: {
    margin: 20,
    width: '80%'
  }
});