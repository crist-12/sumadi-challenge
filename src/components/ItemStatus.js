import { Layout, Text } from '@ui-kitten/components'
import React from 'react'
import { StyleSheet } from 'react-native'
import { Ionicons } from '@expo/vector-icons'

const ItemStatus = ({icon, value, status}) => {

  return (
    <Layout style={styles.textInfo}>
        <Layout style={styles.containerBattery}>
          <Ionicons name={icon} size={28} color="white" />
        </Layout>
        <Layout style={styles.containerBattery}>
          <Text style={styles.batteryText}>
            {value}{status}
          </Text>
        </Layout>
      </Layout>
  )
}

export default ItemStatus;

const styles = StyleSheet.create({
    batteryText: {
      color: 'white',
      fontWeight: "bold"
    },
    textInfo: {
      display: 'flex',
      backgroundColor: 'transparent',
      flexDirection: 'row',
      marginLeft: 10
    },
    containerBattery: {
      backgroundColor: 'transparent',
      display: 'flex',
      justifyContent: 'center',
      marginRight: 5
    }
  });
  