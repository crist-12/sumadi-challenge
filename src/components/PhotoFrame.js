import React from "react";
import { StyleSheet, Image } from "react-native";
import { Layout } from "@ui-kitten/components";
import { usePictureStore } from "../providers/PictureStore";



const PhotoFrame = ({children, heightFrame}) => {

    const photo = usePictureStore((state) => state.photo);

    return (
        <Layout style={[styles.pictureFrame, {flex: heightFrame}]}>
            <Layout style={styles.cameraFrame}>
                {/* <Image style={styles.camera} source={{ uri: "data:image/jpg;base64," + photo }} /> */}
                {children}
            </Layout>
        </Layout>
    )
}

export default PhotoFrame;

const styles = StyleSheet.create({
    pictureFrame: {
        //flex: 4,
        backgroundColor: 'white',
        width: '100%',
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',    
      },
      cameraFrame: {
        display: 'flex',
        width: '80%',
        height: '90%',
        borderStyle: 'dashed',
        borderWidth: 1
      },
      camera: {
        flex: 1,
      },
})