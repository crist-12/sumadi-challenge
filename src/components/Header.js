import {
  View,
  KeyboardAvoidingView,
  StyleSheet,
  StatusBar,
} from "react-native";
import * as Battery from "expo-battery";
import React, { useEffect, useState } from "react";
import { Text, Layout } from "@ui-kitten/components";

import ItemStatus from "./ItemStatus";

import { useBatteryStore } from '../providers/BatteryStore'
import { useNetStore } from '../providers/NetStore'

import NetInfo, { useNetInfo } from "@react-native-community/netinfo";

const Header = () => {

  const isAvailable = useBatteryStore((state) => state.isAvailable);
  const setIsAvailable = useBatteryStore((state) => state.setAvailable);
  const batteryLevel = useBatteryStore((state) => state.batteryLevel);
  const setBatteryLevel = useBatteryStore((state) => state.setBatteryLevel);
  const batteryState = useBatteryStore((state) => state.batteryState);
  const setBatteryState = useBatteryStore((state) => state.setBatteryState);
  const batteryStateDescription = useBatteryStore((state) => state.batteryStateDescription);
  const batteryIcon = useBatteryStore((state) => state.batteryIcon);


  const setIsConnected = useNetStore((state) => state.setIsConnected);
  const netIcon = useNetStore((state) => state.netIcon);
  const netStatus = useNetStore((state) => state.networkStatus);
 


  useEffect(() => {
    
    (async () => {
      const [isAvailable, batteryLevel, batteryState, lowPowerMode] =
        await Promise.all([
          Battery.isAvailableAsync(),
          Battery.getBatteryLevelAsync(),
          Battery.getBatteryStateAsync(),
          Battery.isLowPowerModeEnabledAsync(),
        ]);

      setIsAvailable(isAvailable);
      setBatteryLevel((batteryLevel * 100).toFixed(0));
      setBatteryState(batteryState);
      setIsConnected(NetInfo.isConnected)
    })();
    
    const networkListener = NetInfo.addEventListener(state => {
     setIsConnected(state.isConnected)
    });

    const batteryLevelListener = Battery.addBatteryLevelListener(
      ({ batteryLevel }) => {
        setBatteryLevel((batteryLevel * 100).toFixed(0))
      }
    );
    const batteryStateListener = Battery.addBatteryStateListener(
      ({ batteryState }) => {
        setBatteryState(batteryState)
      }
    );
    const lowPowerModeListener = Battery.addLowPowerModeListener(
      ({ lowPowerMode }) => setLowPowerMode(lowPowerMode)
    );
    
    return () => {
      networkListener
      batteryLevelListener /*&& batteryLevelListener.remove();*/
      batteryStateListener /*&& batteryStateListener.remove();*/
      lowPowerModeListener /*&& lowPowerModeListener.remove();*/
    };
  }, []);



  return (
    <Layout
      style={[
        styles.header,
        { backgroundColor: "white", flex: 1, width: "100%" },
      ]}
    >
      <Layout style={styles.header}>
        <Text style={styles.title}>Sumadi Test App</Text>
       <ItemStatus icon={batteryIcon} value={batteryLevel + "% "} status={batteryStateDescription}/>
      {/* <ItemStatus icon={netInfo.isConnected ? 'checkmark-circle-outline': 'close-circle-outline'} status={netInfo.isConnected ? 'CONNECTED': 'DISCONNECTED'}/> */}
      <ItemStatus icon={netIcon} status={netStatus}/>
      </Layout>
    </Layout>
  );
};

export default Header;

const styles = StyleSheet.create({
  header: {
    flex: 1,
    backgroundColor: "rgba(0,190,240,1)",
    width: "100%",
    justifyContent: "center",
    borderBottomRightRadius: 120,
  },
  title: {
    fontSize: 22,
    color: "white",
    fontWeight: "bold",
    margin: 15,
  },
  batteryInfo: {
    flexDirection: "row",
  },
  batteryText: {
    color: 'white',
    fontWeight: "bold"
  },
  textInfo: {
    display: 'flex',
    backgroundColor: 'transparent',
    flexDirection: 'row',
    marginLeft: 10
  },
  containerBattery: {
    backgroundColor: 'transparent',
    display: 'flex',
    justifyContent: 'center',
    marginRight: 5
  }
});
