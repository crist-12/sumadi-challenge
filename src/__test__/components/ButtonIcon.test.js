import { render, screen, within } from "@testing-library/react-native";
import '@testing-library/react-native'
import * as eva from "@eva-design/eva";
import { ApplicationProvider } from "@ui-kitten/components";
import ButtonIcon from "../../components/ButtonIcon";
import "@testing-library/jest-dom/extend-expect";

describe('It renders', () => {
  it('renders successfully', () => {
    render(<ApplicationProvider {...eva} mapping={eva.mapping} theme={eva.light}>
      <ButtonIcon />
    </ApplicationProvider>);
  })
  it('show the text', () => {
    const testWord = 'Testing';

    const { getByText } =  render(<ApplicationProvider {...eva} mapping={eva.mapping} theme={eva.light}>
      <ButtonIcon text={testWord} style={{flex: 1}} />
    </ApplicationProvider>);

  expect(getByText(testWord)).toBeTruthy();
  })
})

