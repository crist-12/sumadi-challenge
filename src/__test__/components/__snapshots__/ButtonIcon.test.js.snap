// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#3366FF",
        "borderColor": "#3366FF",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "margin": 20,
        "minHeight": 40,
        "minWidth": 40,
        "opacity": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 12,
        "width": "80%",
      }
    }
    testID="buttonIconTest"
  >
    <Text
      appearance="default"
      category="p1"
      style={
        Array [
          Object {
            "color": "#222B45",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
          },
          Object {
            "color": "#FFFFFF",
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginHorizontal": 10,
          },
        ]
      }
    >
      Testing
    </Text>
  </View>
</View>
`;
